### DISCUSSION TITLE
Accessing Route Handlers in Production doesn't work 😟

### DISCUSSION THREAD

**Discussion by hecker** [vercel/next.js#47629] [Help] [answered]
Quick explanation of what I do: I type in a text and the output is an emoji that fits best to the description:

<img width="1512" alt="Screenshot_2023-03-28_at_19 17 01" src="https://user-images.githubusercontent.com/23746655/228364699-bfdef0b6-03e8-4b2d-a42e-3ba40f7410ca.png">

In development mode, everything is okay. But when deploying it, it doesn't work. Maybe there is a problem with how I fetch the API, the security roles, or sth. else?

Here are the important classes (it's open-source):
- [next.config.js](https://github.com/hecker/emoji.cx/blob/main/next.config.js)
- [app/page.tsx](https://github.com/hecker/emoji.cx/blob/main/app/page.tsx)
- [app/api/emoji/route.ts](https://github.com/hecker/emoji.cx/blob/main/app/api/emoji/route.ts)

The status code returned was 400 (in my Safari console). You can check it out yourself on [emoji.cx](https://emoji.cx).

I'd be glad for any help! 👋

**Comment by icyJoseph** [comment, marked as answer]
It could be a few things:

- Do you have the same Node version locally and on the Vercel deployment?
- ~The emoji.json file that's about 1.25MB, and the Code-Size limit for edge functions is 1MB. Source: https://vercel.com/docs/concepts/functions/edge-functions/limitations#code-size-limit~
   - Edit: Nope, this one is not imported on your API route ~ my bad

I'll try to give it a go later on, but you should verify those two .

**Reply by icyJoseph** [reply]
In this branch:

```js
  } catch (e) {
    return new Response("Request cannot be processed!", {
      status: 400,
    });
  }
```

It might be worth logging out the error.

**Reply by icyJoseph** [reply]
@hecker I couldn't resist, so I deployed and did some debugging.

The error is a bug: https://github.com/vercel/next.js/issues/46337 - already fixed in canary :) 

For record's sake, this is the top of the stack you'd see:

```console
Exception: TypeError: Illegal invocation
    at (app/api/emoji/route.ts:12:37)
```

So, you just have to wait for a new release, or take canary.


https://user-images.githubusercontent.com/21013447/228470210-82f59404-4fee-4b65-8508-231ba70ef96a.mov

**Reply by hecker** [reply]
I am totally new to web development, this is beyond my knowledge. I need to see what canary is first.... 😄

But thanks for the tip!

**Reply by icyJoseph** [reply]
That means installing Next.js canary version, which is the one that's build every time the canary branch passes all tests.

> Note that the bug is in Next.js - nothing wrong with your code - This is what we can expect of a beta stage technology such as the appDir

This means that this bug has been fixed, and it is just waiting for a new official version, for example next@13.2.5, since we are currently at `13.2.4`.

To install Next.js canary you can do `pnpm install next@canary`, and it'll do it for you.

However, it might be worth just waiting for the official version, since canary, as you can imagine, is constantly changing.

**Reply by hecker** [reply]
I just wanted to try it out and it worked: https://emoji.cx
But I'll switch back to Next.js 13.2.5 when it's out (and fixed) by then. :)

**Reply by icyJoseph** [reply]
Nice! :) yeah I mean you can keep the deployment up with the next@canary version. Not sure when a new version will release, but at least you have a functioning app now.

Please don't forget to mark as answered 🎉

**Comment by camerondevuk** [comment]
or you should of done npm i next@canary react@latest react-dom@latest eslint-config-next@latest
